common/models/checkpoint.js

```diff
@@ -27,43 +28,59 @@ module.exports = function(Checkpoint) {
    * Get the current checkpoint id
    * @callback {Function} callback
    * @param {Error} err
-   * @param {Number} checkpointId The current checkpoint id
+   * @param {Number} checkpoint The current checkpoint seq
    */
-
   Checkpoint.current = function(cb) {
     var Checkpoint = this;
-    this.find({
-      limit: 1,
-      order: 'seq DESC'
-    }, function(err, checkpoints) {
-      if (err) return cb(err);
-      var checkpoint = checkpoints[0];
-      if (checkpoint) {
-        cb(null, checkpoint.seq);
-      } else {
-        Checkpoint.create({ seq: 1 }, function(err, checkpoint) {
-          if (err) return cb(err);
-          cb(null, checkpoint.seq);
-        });
+
+    async.waterfall([
+      function(cb) {
+        Checkpoint.findOne(cb);
+      },
+      function(cp, cb) {
+        if (cp) {
+          cb(null, cp.seq);
+        } else {
+          Checkpoint.init(function(err, data) {
+            cb(err, data.seq);
+          });
+        }
       }
-    });
+    ],
+    cb);
```

Comment:
This particular part is pretty simple and can be coded without `async`. Keep in mind that `async` adds a bit of overhead, so it's better to not overuse it.

Could you please rewrite this without `async`?